From 7781406a36330be0f4a7e8a0c4dbed51c36cb422 Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Tue, 18 Mar 2025 09:30:24 +0100
Subject: [PATCH 813/954] feat(stm32mp2-fdts): add fixed IO compensation on
 vddio regulators for stm32mp251

IO compensation could alter duty cycle of high frequency output signal
on stm32mp25 SoC. To avoid such issue, fixed IO compensation is used.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 9235204336d0f8202972fbe4c767876d6ea30156 feat(stm32mp2-fdts): add fixed IO compensation on vddio regulators for stm32mp251]
Change-Id: I5501dbf65d762488d8c75ed18cb7d7c5aafad86c
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/444756
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
---
 fdts/stm32mp251.dtsi | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/fdts/stm32mp251.dtsi b/fdts/stm32mp251.dtsi
index dcf1a9963b..66124c4eb5 100644
--- a/fdts/stm32mp251.dtsi
+++ b/fdts/stm32mp251.dtsi
@@ -409,22 +409,27 @@
 
 			vddio1: vddio1 {
 				regulator-name = "vddio1";
+				st,iocomp = <8 7>;
 			};
 
 			vddio2: vddio2 {
 				regulator-name = "vddio2";
+				st,iocomp = <8 7>;
 			};
 
 			vddio3: vddio3 {
 				regulator-name = "vddio3";
+				st,iocomp = <8 7>;
 			};
 
 			vddio4: vddio4 {
 				regulator-name = "vddio4";
+				st,iocomp = <8 7>;
 			};
 
 			vddio: vddio {
 				regulator-name = "vddio";
+				st,iocomp = <8 7>;
 			};
 		};
 
-- 
2.53.0

